pulsar-client-cpp/lib/MultiTopicsConsumerImpl.cc

```diff
@@ -86,11 +86,14 @@ void MultiTopicsConsumerImpl::handleOneTopicSubscribed(Result result, Consumer c
 
     if (result != ResultOk) {
         state_ = Failed;
+        // Use the first failed result
+        auto expectedResult = ResultOk;
+        failedResult.compare_exchange_strong(expectedResult, result);
```

Comment:
Why is `failedResult` a member variable, can you directly set the `result` to line 107? I haven't seen `failedResult` used elsewhere.

Reply:
No. line 107 is in the block of `if (topicsNeedCreate->load() == 0)`, which means only the last time `handleOneTopicSubscribed` is called. Usually, when multiple topics failed to subscribe, the 1st topic would fail with the real reason, but the following topics would fail with `ResultAlreadyClosed`.

Reply:
Get it, thanks.